src/globals.c

```diff
@@ -72,72 +61,159 @@ char *elf_strtab;
 char *elf_section;
 
 /**
- * insert_trie() - Inserts a new element into the trie structure.
- * @trie: A pointer to the trie where the name will be inserted.
- * @name: The name to be inserted into the trie.
- * @funcs_index: The index of the pointer to the func_t. The index is recorded
- *     in a 1-indexed format. Because the first element of 'FUNCS' has been
- *     reserved, there is no need to shift it.
- * Return: The index of the pointer to the func_t.
+ * hash_fnv1a() - hashes a string with FNV-1a hash function.
```

Comment:
Improve the comments for hash function evaluation and considerations.

Reply:
Since the optimizer lacks of the ability to inline functions, you can simply move the body of `hash_fnv1a` into `hash_index`.